common/xbps-src/shutils/pkgtarget.sh

```diff
@@ -1,5 +1,24 @@
 # vim: set ts=4 sw=4 et:
 
+check_existing_pkg() {
+    if [ -z "$XBPS_PRESERVE_PKGS" ]; then
+        return
+    fi
+    if [ -n "$XBPS_TARGET_MACHINE" ]; then
+        arch=$XBPS_TARGET_MACHINE
+    else
+        arch=$XBPS_MACHINE
+    fi
+    if [ -z "$XBPS_CROSS_BUILD" -a -n "$XBPS_ARCH" -a "$XBPS_ARCH" != "$XBPS_TARGET_MACHINE" ]; then
+        arch=${XBPS_ARCH}
+    fi
```

Comment:
Shouldn't `XBPS_TARGET_MACHINE` always be set? We use it in templates as though it's always set...